File: CollegeDatabase/Persons/Professor.cs
using System;

namespace CollegeDatabase.Persons
{
    public class Professor : AbstractPerson
    {
        // Dodatkowe pole dla Profesora
        public string Title { get; set; }

        public Professor(string firstName, string lastName, string title)
        {
            this.Id = Guid.NewGuid();
            this.FirstName = firstName;
            this.LastName = lastName;
            this.Title = title;
        }

        public override string ToString()
        {
            return $"Professor[Id = { this.Id }, First Name = { this.FirstName }, Last Name { this.LastName }, Title = { this.Title }]";
        }
    }
}

File: CollegeDatabasePython/Professor.py
from AbstractPerson import AbstractPerson

class Professor(AbstractPerson):
    
    def __init__(self, firstName, lastName, title):
        AbstractPerson.__init__(self, firstName, lastName)
        self.Title = title

    def Title(self):
        return self.Title

    def ToString(self):
        return "Professor[Id = {0}, First Name = {1}, Last Name {2}, Title = {3}]".format(self.Id, self.FirstName, self.LastName, self.Title);